src/platform/SIMULATOR/platform_mcu.h

```diff
@@ -30,3 +30,7 @@
 #define IOCFG_IN_FLOATING   0
 
 #define SPIDEV_COUNT        0
+
+// no serial pins are defined for the simulator
+#undef SERIAL_TRAIT_PIN_CONFIG
```

Comment:
Maybe make it work without common_pre.h too:
```suggestion
#ifdef SERIAL_TRAIT_PIN_CONFIG
#undef SERIAL_TRAIT_PIN_CONFIG
#endif
```